module-2/backend/week_one.md

```diff
@@ -5,30 +5,100 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+ 
 6. What types of variables are accessible in our view templates without explicitly passing them?
+
+ params
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
+  @count = 1
   get '/horses' do
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+
+  ```ruby
+  @count = 1
+  params[name]="Mr.Ed"
+  get '/horses' do
+    erb :index
+  end
+  ```
 9. What's the purpose of ERB?
+
+ ERB is a template language that can produce static pages/iterate through Ruby. It helps reduce redundancy of written HTML.
+
 10. Why do I need a development AND test database?
+
+ Develpment and test database must be unique to their environment as information stored in one will differ from the other, due to the separate functions of each environment.
+
 11. What's responsive design?
+
+ Mobile-oriented design that responds to changing the browswer size.
+
 12. What is CRUD and why is it important?
+
+ CRUD is create, read, update, delete. It is a cornerstone of functionality in web development.
+
 13. What does HTTP stand for? 
+
+ Hyper-text transfer protocol.
+
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+ <% Ruby %> and <%= Ruby %>. The first will simply run the ruby code while the latter will display the output.
+
 15. What's an ORM?
+
+ An ORM is an object relational mapper. It writes in object oriented language wrapped around a relational database. Obj classes are mapped to tables in the database.
+
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+
+ ActiveRecord
+
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+- get '/restaurants/ Displays index of restauraunts
+- get '/restauraunts/new'Displays form
+- post '/restauraunts'Creates new restauraunt in the database
+- get '/restauraunts/:id/edit'Displays a form
+- put '/restauraunts/:id'Updates database
+- get '/restauraunts/:id' Displays a single instance of restauraunt
+- delete '/restauraunts/:id' Deletes from database
+
 18. What's a migration? 
+
+ A migration creates the structure of the database by adding or removing tables.
+
 19. When you create a migration, does it automatically modify your database?
+
+ No. The migration must be migrated, which creates the schema and modifies the database.
+
 20. How does a model relate to a database?
+
+ A model defines an object and it's methods within a table of the database.
+
 21. What's the difference between agile workflow and waterfall method?
+
+ I know what agile workflow is but can't remember waterfall. Waterfall is top down approach to agile workflow or something like that?

```

Comment:
Waterfall groups steps so that you do all of your planning up front and then all coding and then all testing and then all collection of feedback. Basically it means that you have a full blown application before you've ever really gotten any feedback. Agile allows you to get feedback on smaller functional chunks as you build.